target/linux/ramips/dts/rt5350_zyxel_keenetic-lite_rev.b.dts

```diff
@@ -0,0 +1,121 @@
+/dts-v1/;
+
+#include "rt5350.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "zyxel,keenetic-lite_rev.b", "ralink,rt5350-soc";
+	model = "ZyXEL Keenetic Lite rev.B";
+
+	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status: power {
+			label = "zyxel:green:power";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
+
+		wps {
+			label = "zyxel:green:wps";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+
+		internet {
+			label = "zyxel:green:internet";
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan {
+			label = "zyxel:green:wlan";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 0 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can this be increased?

Reply:
I increased it to 40,000,000 and install the image, it works fine, I did not notice any changes in the speed of work.

Reply:
This can be tested by reading a defined amount of data from flash (e.g. `dd if=/dev/mtdX of=/dev/null bs=2M count=1`)

Despite, it might also be a good idea to check the maximum speed in the flash specifications.

Reply:
I can not find the maximum flash speed in the specification. Searched [here](https://cdn.sparkfun.com/datasheets/Wireless/WiFi/RT5350.pdf) and [here](http://read.pudn.com/downloads622/ebook/2530552/Ralink_SDK_User_Manual_4.2.pdf). Experimentally found the maximum speed (60,000,000). With a further increase, the device does not boot.
At 10,000,000, the command `time dd if=/dev/mtdblock6 of=/dev/null bs=10M count=1` is executed in 8.3s
at 40,000,000 in 4.01s
at 60,000,000 in 3.5s

Reply:
So you mean `spi-max-frequency = <60000000>;` worked? Interesting, I thought that frequency is too high for RT5350.
BTW you can check the exact hardware frequency by changing this line from dev_dbg to dev_info.
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/patches-4.14/0042-SPI-ralink-add-Ralink-SoC-spi-driver.patch#L234

Reply:
When I change this line `dmesg` shows the frequency that I set.
`root@OpenWrt:~# dmesg | grep spi`
`[    0.607320] spi spi0.0: force spi mode3`
`[    0.615666] m25p80 spi0.0: speed: 60000000/60000000, rate: 2, prescal: 0, loops: 130` #60000000/**60000000** - this speed is always the same as I set in the device tree file
`[    0.632101] m25p80 spi0.0: mx25l6405d (8192 Kbytes)`
`[    0.642048] 4 fixed-partitions partitions found on MTD device spi0.0`
`[    0.654791] Creating 4 MTD partitions on "spi0.0":`

Reply:
Good to hear that. I've never seen a device with rate 2 working, but this device seems to work.